tests/integration/suite/test_node.py

```diff
@@ -145,3 +147,22 @@ def wait_for_cloud_credential(client, cloud_credential_id, timeout=60):
             if val["id"] == cloud_credential_id:
                 cred = val
     return cred
+
+
+def wait_for_node_template(client, node_template_id, timeout=60):
+    start = time.time()
+    interval = 0.5
+    nodeTemplates = client.list_node_template()
```

Comment:
I think we need to list inside while loop, so we keep getting nodetemplates after some interval?